src/main/java/com/epam/DAO/AmmunitionDAOImpl.java

```diff
@@ -0,0 +1,24 @@
+package com.epam.DAO;
+
+import com.epam.entity.Ammunition;
+import com.epam.service.InputServiceImpl;
+
+import java.util.*;
+
+public class AmmunitionDAOImpl implements AmmunitionDAO {
+
+    public Set<Ammunition> getAll() {
+        Set<Ammunition> ammun = new HashSet<Ammunition>();
+        ammun.addAll(new WeaponDAOImpl().getAll());
```

Comment:
Dao could be heavy object. Better make factory with singleton that returns instance

Reply:
In your case it is real connecrion to DB. Heavy thing